cpc/sqlx/cpc_sketch_get_estimate.sqlx

```diff
@@ -30,13 +30,13 @@ Param seed: This is used to confirm that the given sketch was configured with th
 Returns: a FLOAT64 value as the cardinality estimate.
 For more details: https://datasketches.apache.org/docs/CPC/CPC.html'''
 ) AS R"""
-const default_seed = BigInt(Module.DEFAULT_SEED);
 try {
-  var sketch = Module.cpc_sketch.deserializeFromB64(sketch, seed ? BigInt(seed) : default_seed);
+  var sketchObject = null;
   try {
-    return sketch.getEstimate();
+    sketchObject = Module.cpc_sketch.deserializeFromB64(sketch, seed ? BigInt(seed) : BigInt(Module.DEFAULT_SEED));
+    return sketchObject.getEstimate();
   } finally {
-    sketch.delete();
+    if (sketchObject != null) sketchObject.delete();
   }
 } catch (e) {
   throw new Error(Module.getExceptionMessage(e));
```

Comment:
Are you concerned that the sketchObject.delete() would throw an exception? 
Otherwise a try ... catch ... finally block would be easier to read.

Reply:
well, perhaps in some cases the inner and outer blocks can be combined